### ISSUE TITLE
app crash on basic operations

### ISSUE THREAD

**Issue by jfcolom** [nicolasbrailo/VlcFreemote#3]
The app properly connects to VLC service and let me to browse media in server. When I select some media file it starts playing. However, after pausing/resuming or changing the volume two or three times, the app crashes. Unfortunately, I cannot find an exact failure pattern, it seems quite random.

``logcat``:

```
09-02 19:42:14.867 28209 28209 W IInputConnectionWrapper: finishComposingText on inactive InputConnection
09-02 19:42:14.869 28209 28209 W IInputConnectionWrapper: reportFullscreenMode on inexistent InputConnection
09-02 19:42:14.869 28209 28209 W IInputConnectionWrapper: finishComposingText on inactive InputConnection
```

The app stops each time I try to start it up, until I restart VLC service.

VLC version:
```
VLC media player 3.0.3 Vetinari (revision 3.0.2-225-gc9e3360dd4)
```

**Comment by nicolasbrailo** [comment]
Thanks for the report @jfcolom! I can't reproduce this issue in any of my test devices, so I'd love your help to figure out what's going on.

Besides the "InputConnection" messages, have you seen a stacktrace in `adb logcat`? It should look something like what I'm copying below. That would help me by marking the exact point of failure. Also, which Android version are you currently using?

Thanks!


Stacktrace example:

> --------- beginning of crash
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: FATAL EXCEPTION: main
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: Process: com.nicolasbrailo.vlcfreemote, PID: 1729
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: java.lang.RuntimeException: HOLA
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.ServerSelectView.toggleServerScanning(ServerSelectView.java:185)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.ServerSelectView.onClick(ServerSelectView.java:94)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.view.View.performClick(View.java:6261)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.widget.TextView.performClick(TextView.java:11159)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:23748)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:751)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6776)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1496)
> 09-03 14:11:20.419  1729  1729 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1386)
>

**Comment by jfcolom** [comment]
I'm sorry, I'm not very experienced extracting logs in Android. I don't use ``adb``, but I run ``logcat`` directly on a terminal app in my smartphone. I have just discovered that  I'm getting much more messages if I run ``logcat`` as root. After filtering the output, I've found the following relevant messages and stacktrace:

```
09-03 17:32:12.950   753 25755 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.nicolasbrailo.vlcfreemote/.MainActivity bnds=[804,475][1068,860] (has extras)} from uid 10032 on display 0

09-03 17:32:12.994   753  2552 I ActivityManager: Start proc 26303:com.nicolasbrailo.vlcfreemote/u0a75 for activity com.nicolasbrailo.vlcfreemote/.MainActivity
09-03 17:32:13.055 26303 26303 W System  : ClassLoader referenced unknown path: /data/app/com.nicolasbrailo.vlcfreemote-1/lib/arm

09-03 17:32:13.276   753  1194 I ActivityManager: Displayed com.nicolasbrailo.vlcfreemote/.MainActivity: +296ms (total +42s925ms)

09-03 17:32:16.561 26303 26303 W com.nicolasbrailo.vlcfreemote.net_utils.ServerScanner: IP FE80::DC0F:5AFF:FEC2:5BD1%DUMMY0 is not of a supported type (IPv6?)
09-03 17:32:16.561 26303 26303 W com.nicolasbrailo.vlcfreemote.net_utils.ServerScanner: IP FE80::B69D:BFF:FE46:D5D0%WLAN0 is not of a supported type (IPv6?)
09-03 17:32:16.604 26303 26303 E         : Couldn't `cd /home/student/mirror`. Will `cd ~` instead.

09-03 17:32:46.913 26303 26319 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
09-03 17:32:46.913 26303 26319 E AndroidRuntime: Process: com.nicolasbrailo.vlcfreemote, PID: 26303
09-03 17:32:46.913 26303 26319 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at android.os.AsyncTask$3.done(AsyncTask.java:325)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:761)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: Caused by: java.lang.NumberFormatException: For input string: "4401716775"
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.lang.Integer.parseInt(Integer.java:524)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.lang.Integer.parseInt(Integer.java:556)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.vlc_connector.VlcCommand_ReturnsVlcStatus$1$1.parseValue(VlcCommand_ReturnsVlcStatus.java:47)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.vlc_connector.VlcCommand_ReturnsVlcStatus$1$1.parseValue(VlcCommand_ReturnsVlcStatus.java:42)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.vlc_connector.http_utils.XmlObjectReader.xmlParseImpl(XmlObjectReader.java:41)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.vlc_connector.http_utils.XmlMogrifier.doInBackground(XmlMogrifier.java:80)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at com.nicolasbrailo.vlcfreemote.vlc_connector.http_utils.XmlMogrifier.doInBackground(XmlMogrifier.java:20)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at android.os.AsyncTask$2.call(AsyncTask.java:305)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
09-03 17:32:46.913 26303 26319 E AndroidRuntime: 	... 4 more
09-03 17:32:46.915   753  4153 W ActivityManager:   Force finishing activity com.nicolasbrailo.vlcfreemote/.MainActivity
```

"NumberFormatException" suggests to me that the app is getting some unexpected input from VLC server. I attach a sample of [status.xml](https://github.com/nicolasbrailo/VlcFreemote/files/2346145/status.txt) (renamed to status.txt) while playing a file, so you can see an example of the numbers provided my server.

It seems to me that the problem has to do more with my VLC server version, than with my Android version. Anyway, I post the two versions:

- VLC version: 3.0.3 Vetinari (3.0.2-225-gc9e3360dd4)
- Android version: 7.1.2

Please, let me know if I can provide you any other useful data or do some test. Good luck!

**Comment by jfcolom** [comment]
Confirmed! My VLC installation was wrong (precompiled LUA scripts from previous version, not properly generated for new version).

Thank you

**Comment by nicolasbrailo** [comment]
Thanks for not giving up @jfcolom! Most people would have said "doesn't work" and just uninstalled the app.

The stacktrace you posted is exactly what I was looking for. Seems VLC is sending a crazy value as the length of a media item (I'm betting you don't have a 140 years long video in there). Now, the app shouldn't crash - it should just refuse to load that the media item, so I'll be fixing this bug soon. The information you posted was really helpful in troubleshooting this!

Just so that I'm clear: With a 'vanilla' VLC version you don't experience any other issues and VlcFreemote is now usable without any crashes. Any fixes I release will be just to improve the app, but are not strictly required to get you up&running. Is that correct?

**Comment by jfcolom** [comment]
Yes it is correct. VlcFreemote 1.6 is fully usable according to my tests.

I wish long life to VlcFreemote, thank you for your work in future improvements.

**Comment by nicolasbrailo** [comment]
Excellent! In the meantime I released version 1.7, which:
- shouldn't crash on invalid status anymore
- should make it easier to create bug reports
I'm not sure how the update process goes in F-Droid, so it may take a while until it makes it there.

As the only user who has provided feedback so far: If you encounter any problems, have any feature requests or usability suggestions, please let me know!

**Comment by jfcolom** [comment]
Ok @nicolasbrailo , I will input some issues prefixed as "Feature request" with some suggestions for your consideration. I hope they have a good coding effort-benefit relationship, and you or somebody else in the community find some time to implement them, while keeping VlcFreemote reasonably simple.
